src/tsit5/atsit5.jl

```diff
@@ -349,18 +360,245 @@ function DiffEqBase.step!(integ::SAT5I{false, T, S}) where {T, S}
+
+    k1, k2, k3, k4, k5, k6, k7 = integ.ks
+    tmp = integ.tmp; f! = integ.f
+
+    @inbounds for j in 1:M
+        integ.uprev[j] .= integ.u[j]
+    end
+    uprev = integ.uprev; u = integ.u
+
+    qold = integ.qold; abstol = integ.abstol; reltol = integ.reltol
+
+    # FSAL
+    if integ.u_modified
+        f!(k1, uprev, p, t)
+        integ.u_modified=false
+    else
+        @inbounds for j in 1:M; k1[j] .= k7[j]; end
+    end
+
+    EEst = Inf
+
+    @inbounds while EEst>1
+        dt < 1e-14 && error("dt<dtmin")
+
+        for j in 1:M
+            for i in 1:L
+                tmp[j][i] = uprev[j][i]+dt*a21*k1[j][i]
+            end
+        end
+
+        f!(k2, tmp, p, t+c1*dt)
+        for j in 1:M
+            for i in 1:L
+                tmp[j][i] = uprev[j][i]+dt*(a31*k1[j][i]+a32*k2[j][i])
+            end
+        end
+
+        f!(k3, tmp, p, t+c2*dt)
+        for j in 1:M
+            for i in 1:L
+                tmp[j][i] = uprev[j][i]+dt*(a41*k1[j][i]+a42*k2[j][i]+a43*k3[j][i])
+            end
+        end
+
+        f!(k4, tmp, p, t+c3*dt)
+        for j in 1:M
+            for i in 1:L
+                tmp[j][i] = uprev[j][i]+dt*(a51*k1[j][i]+a52*k2[j][i]+a53*k3[j][i]+a54*k4[j][i])
+            end
+        end
+
+        f!(k5, tmp, p, t+c4*dt)
+        for j in 1:M
+            for i in 1:L
+                tmp[j][i] = uprev[j][i]+dt*(a61*k1[j][i]+a62*k2[j][i]+a63*k3[j][i]+a64*k4[j][i]+a65*k5[j][i])
+            end
+        end
+
+        f!(k6, tmp, p, t+dt)
+        for j in 1:M
+            for i in 1:L
+                u[j][i] = uprev[j][i]+dt*(a71*k1[j][i]+a72*k2[j][i]+a73*k3[j][i]+a74*k4[j][i]+a75*k5[j][i]+a76*k6[j][i])
+            end
+        end
+
+        f!(k7, u, p, t+dt)
+        for j in 1:M
+            for i in 1:L
+                tmp[j][i] = dt*(btilde1*k1[j][i]+btilde2*k2[j][i]+btilde3*k3[j][i]+btilde4*k4[j][i]+
+                            btilde5*k5[j][i]+btilde6*k6[j][i]+btilde7*k7[j][i])
+                tmp[j][i] = tmp[j][i]/(abstol+max(abs(uprev[j][i]),abs(u[j][i]))*reltol)
+            end
+        end
+
+        EEst = integ.internalnorm(tmp)
+
+        if iszero(EEst)
+            q = inv(qmax)
+        else
+            @fastmath q11 = EEst^beta1
+            @fastmath q = q11/(qold^beta2)
+        end
+
+        if EEst > 1
+            dt = dt/min(inv(qmin),q11/gamma)
+        else # EEst <= 1
+            @fastmath q = max(inv(qmax),min(inv(qmin),q/gamma))
+            qold = max(EEst,qoldinit)
+            dtold = dt
+            dt = dt/q #dtnew
+            dt = min(abs(dt),abs(tf-t-dtold))
+
+            integ.dt = dt
+            integ.qold = qold
+            integ.tprev = t
+
+            if (tf - t - dtold) < 1e-14
+                integ.t = tf
+            else
+                integ.t += dtold
+            end
+        end
+    end
+    return  nothing
+end
+
+# Vector{SVector}
+function DiffEqBase.step!(integ::SAT5I{true, T, S}) where {T, S<:Vector{<:SVector}}
```

Comment:
is this not the exact same code as the out of place one? Or just one if statement away?

Reply:
The Vector{SVector} method works properly. Which line are you referring to? (which if statement) ?

I have problems with the method of Vector{Vector} though. It doesn't work at all!

Reply:
The Vector{Vector} one is probably just aliasing because I don't see a `recursivecopy` in there. This one works but I am not sure this code is even necessary? If there is code necessary here, it doesn't need a whole dispatch, just 1 or 2 lines added to the scalar one.

Reply:
Yeah you might be right! Where do I grab `recursivecopy` from? Also, do you have any idea what should be `recursivecopy`'d ?

> This one works but I am not sure this code is even necessary? If there is code necessary here, it doesn't need a whole dispatch, just 1 or 2 lines added to the scalar one.

You also could be right here, but I don't see immediately how to do it... I somehow have to get rid of all of `i in 1:M` loops for what you say to work.

Reply:
Anything that is a Vector{Vector} needs to be recursive copied or they will alias. It's in RecursiveArrayTools.jl. You could also deepcopy but that's super slow, whereas recursivecopy is copy but made recursive for Vector{Vector} structures (and handles the Vector{Number} and Number case too, so you can just use it everywhere). 

>You also could be right here, but I don't see immediately how to do it... I somehow have to get rid of all of i in 1:M loops for what you say to work.

Vector arithmetic with SVectors should just work?

Reply:
You were absolutely correct, using `deepcopy` on `u0` solved the issue. I'll use RecursiveArrayTools (if you are ok as adding it as a dependency) and then look into removing the method for Vector{SVector}.

Reply:
RecursiveArrayTools is already an indirect dependency to everything diffeq. Just use it.